### PR TITLE
Zpi 90

### PR DESCRIPTION
**Pull request by SzymekLop**

### CODE REVIEW

File: src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+
+    public String toString(){
+        return startDateTime.toString() + "-" + endDateTime.toString() + ": " + this.amount;
+    }
+
+    public ScheduleSlot[] split(LocalDateTime splitBy) {
+        if(!startDateTime.isBefore(splitBy) || !endDateTime.isAfter(splitBy)) {
+            throw new IllegalArgumentException();
```
Comment (ReConsolidated) [inline src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java:55]: add an explanation in the exception message

File: src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof ScheduleSlot that)) return false;
+
+        if (!amount.equals(that.amount)) return false;
+        if (!itemsAvailability.equals(that.itemsAvailability)) return false;
+//        if (itemsAvailability.size() != that.itemsAvailability.size()) return false;
+//        for(int i = 0; i < itemsAvailability.size(); i++) {
```
Comment (ReConsolidated) [inline src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java:113]: commented code

File: src/main/java/io/github/reconsolidated/zpibackend/utils/BooleanListToStringConverter.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@Converter
+public class BooleanListToStringConverter implements AttributeConverter<List<Boolean>,String> {
+
+    private final static String SEPARATOR = ",";
+    @Override
+    public String convertToDatabaseColumn(List<Boolean> attribute) {
+        return attribute == null ? null : StringUtils.join(attribute,SEPARATOR);
+    }
+
+        @Override
```
Comment (ReConsolidated) [inline src/main/java/io/github/reconsolidated/zpibackend/utils/BooleanListToStringConverter.java:22]: this indentation looks weird, did linter accept that?

File: src/main/java/io/github/reconsolidated/zpibackend/features/item/ItemRepository.java [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import java.util.List;
 
 @Repository
+public
```
[the remark this answers is not in the document]
Comment (SzymekLop) [reply src/main/java/io/github/reconsolidated/zpibackend/features/item/ItemRepository.java:9]: I use it in ReservationService from different package

File: src/main/java/io/github/reconsolidated/zpibackend/utils/BooleanListToStringConverter.java (hunk 3 above)
Comment (SzymekLop) [reply src/main/java/io/github/reconsolidated/zpibackend/utils/BooleanListToStringConverter.java:22]: yes

File: src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ReservationService.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import lombok.AllArgsConstructor;
+
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class ReservationService {
+
+    private ReservationRepository reservationRepository;
+    private ItemRepository itemRepository;
```
[the remark this answers is not in the document]
Comment (SzymekLop) [reply src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ReservationService.java]: swped to service

File: src/main/java/io/github/reconsolidated/zpibackend/features/item/ItemRepository.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (SzymekLop) [reply src/main/java/io/github/reconsolidated/zpibackend/features/item/ItemRepository.java]: deleted

File: src/main/java/io/github/reconsolidated/zpibackend/features/reservation/Schedule.java [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+    public void addSlot(ScheduleSlot scheduleSlot) {
+//        CoreConfig coreConfig = item.getStore().getStoreConfig().getCore();
+
+        setSlotType(scheduleSlot);
+
+        ScheduleSlot slotToAdd = scheduleSlot;
+
+        for (int i = 0; i < scheduleSlots.size(); i++) {
+//            if (!coreConfig.getGranularity()) {
```
[the remark this answers is not in the document]
Comment (SzymekLop) [reply src/main/java/io/github/reconsolidated/zpibackend/features/reservation/Schedule.java]: yes

File: src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java (hunk 1 above)
Comment (SzymekLop) [reply src/main/java/io/github/reconsolidated/zpibackend/features/reservation/ScheduleSlot.java]: description added

**Pull request merged**